ui/base-tool/src/main/java/org/cru/godtools/base/tool/activity/BaseToolActivity.kt

```diff
@@ -192,16 +190,23 @@ abstract class BaseToolActivity @JvmOverloads constructor(
     @BindView(R2.id.mainContent)
     internal var mainContent: View? = null
 
-    @CallSuper
-    protected open fun updateVisibilityState() {
-        val state = determineActiveToolState()
-        loadingContent?.visibility = if (state == STATE_LOADING) View.VISIBLE else View.GONE
-        mainContent?.visibility = if (state == STATE_LOADED) View.VISIBLE else View.GONE
-        missingContent?.visibility =
-            if (state == STATE_NOT_FOUND || state == STATE_INVALID_TYPE) View.VISIBLE else View.GONE
+    enum class ToolState {
+        LOADING, LOADED, NOT_FOUND, INVALID_TYPE;
+
+        companion object {
+            // TODO: this should be an actual state
```

Comment:
this will be addressed in a future PR when I also implement an `OFFLINE` state. I just wanted to start utilizing `UNKNOWN` for now with the same behavior as `LOADING`